docs/sphinxext/mantiddoc/directives/amalgamate.py

```diff
@@ -0,0 +1,57 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantiddoc.directives.base import BaseDirective
+
+import os
+import sys
+
+
+class CompilationDirective(BaseDirective):
+
+    """
+    Obtains the summary for a given algorithm based on it's name.
+    """
+
+    required_arguments, optional_arguments = 1, 1
+
+    def run(self):
+        """
+        The main entry point that docutils calls.
+        It calls self.execute to do the main work.
+        Derived classes should override execute() and insert
+        whatever rst they require with self.add_rst()
+        """
+        nodes = self.execute()
+        if self.rst_lines is not None:
+            self.commit_rst()
+        return nodes
+
+    def execute(self):
+        """
+        Called by Sphinx when the ..compiler:: directive is encountered.
+        """
+        args = self.arguments[0]
+        current_dir = os.path.split(sys.argv[0])[0]
+        path_to_notes = '../../../../../../../../docs/source/' + self.source().rsplit('/', 1)[0] + args
```

Comment:
Is there a way we can use Git Python to get the root of the repo like this rather than having to use relative paths everywhere like this:
https://github.com/mantidproject/pre-commit-hooks/blob/main/cmake_missing_pytest_files/check_for_missing_py_tests.py#L25